extensions/wikia/AdEngine/js/provider/gpt/sourcePointTag.js

```diff
@@ -36,16 +73,70 @@ define('ext.wikia.adEngine.provider.gpt.sourcePointTag', [
 			window.dispatchEvent(spReadyEvent);
 		});
 
-		doc.addEventListener('sp.blocking', function () {
+		doc.addEventListener('sp.blocking', (function () {
+			log(['sp.blocking'], 'debug', logGroup);
 			adTracker.track('sourcepoint/blocked');
-		});
+
+			doc.body.classList.add('source-point');
+			blocking = true;
+			this.cmdQueue.start();
+		}).bind(this));
+
+		doc.addEventListener('sp.not_blocking', (function () {
+			log(['sp.not_blocking'], 'debug', logGroup);
+			adTracker.track('sourcepoint/not_blocked');
+
+			this.cmdQueue.start();
+		}).bind(this));
 
 		log('Appending GPT script to head', 'debug', logGroup);
 		node.parentNode.insertBefore(gads, node);
 
 		this.enableServices();
+
 		this.initialized = true;
 	};
 
+	SourcePointTag.prototype.push = function (callback) {
+		log(['push', 'cmdQueue', callback], 'debug', logGroup);
+		this.cmdQueue.push(callback);
+	};
+
+	SourcePointTag.prototype.addSlot = function (adElement) {
+		log(['adSlot', adElement], 'debug', logGroup);
+		var slot = GoogleTag.prototype.addSlot.call(this, adElement);
+
+		if (blocking) {
+			log(['addSlot', 'blocked', adElement.getId()], 'debug', logGroup);
+			slot.setCollapseEmptyDiv(true, true);
+		}
+
+		return slot;
+	};
+
+	SourcePointTag.prototype.onAdLoad = function (slotName, element, gptEvent, onAdLoadCallback) {
+		log(['onAdLoad', slotName], 'debug', logGroup);
+		var iframe,
+			newSlotName,
+			slotElementId;
+
+		if (blocking && gptEvent.slot && gptEvent.slot.getSlotElementId) {
+			slotElementId = gptEvent.slot.getSlotElementId() || element.getId();
+			newSlotName = slotElementId.replace(/(.*)([\/])([^\/]*$)/, '$3');
+
+			if (slotName !== newSlotName) {
+				copyStyles(slotName, newSlotName);
```

Comment:
I'm thinking of gathering slots in an array and copy those styles in one loop. But it's impossible since we do it on an ad-load event and we don't know here if there are more slots to checked or not :/ Any other ideas how to make it better performance-wise? If not, then we'll have to live with it, I guess.